tests/fetchODCDataFilesUtilsTests.js

```diff
@@ -0,0 +1,188 @@
+https://raw.githubusercontent.com/inclusive-design/data-update-github/main/LICENSE
+*/
+
+"use strict";
+
+const fluid = require("infusion");
+const jqUnit = fluid.require("node-jqunit", require, "jqUnit");
+const fs = require("fs");
+const nock = require("nock");
+const rimraf = require("rimraf");
+
+const utils = require("../scripts/fetchODCDataFilesUtils.js");
+
+jqUnit.module("Fetch ODC data files tests");
+
+const odsHostname = "https://data.ontario.ca";
+const odsPath = "/dataset/covid-19-assessment-centre-locations";
+const fullSitePath = odsHostname + odsPath;
+
+//****************** Test getDataSource() ******************
+
+const getDataSourceTestCases = {
+	hasCsv: {
+		message: "The response contains a download link pointing to a csv format: download link and the published date is returned",
+		nockConfig: {
+			status: 200,
+			response: fs.readFileSync(__dirname + "/data/responseWithCsv.html")
+		},
+		expected: {
+			downloadUrl: "https://data.ontario.ca/dataset/8ba078b2-ca9b-44c1-b5db-9674d85421f9/resource/c60993bb-3988-4648-9be9-398dee480514/download/assessment_centre_locations.csv",
+			publishedDate: "2020-11-17"
+		}
+	},
+	noCsv: {
+		message: "The response doesn't contain a download link pointing to a csv format: download link and the published date returns undefined",
+		nockConfig: {
+			status: 200,
+			response: fs.readFileSync(__dirname + "/data/responseWithoutCsv.html")
+		},
+		expected: {
+			downloadUrl: undefined,
+			publishedDate: undefined
+		}
+	},
+	serverDown: {
+		message: "No respose from the server",
+		nockConfig: {
+			status: 400,
+			response: ""
+		},
+		expected: {
+			downloadUrl: undefined,
+			publishedDate: undefined
+		}
+	}
+};
+
+fluid.each(getDataSourceTestCases, function (oneCase, key) {
+	jqUnit.asyncTest("Test getDataSource() - " + key, function () {
+		jqUnit.expect(1);
+		// setup nock to mock the response from the data source
+		nock(odsHostname).get(odsPath).reply(oneCase.nockConfig.status, oneCase.nockConfig.response);
+
+		utils.getDataSource(fullSitePath).then(function (result) {
+			jqUnit.assertDeepEq(oneCase.message, oneCase.expected, result);
+			jqUnit.start();
+		});
+	});
+});
+
+// ****************** Test formatDate() ******************
+
+jqUnit.test("Test formatDate()", function () {
+	const testCases = {
+		format1: {
+			message: "Convert the input format of November 17, 2020",
+			input: "November 17, 2020",
+			expected: "2020-11-17"
+		},
+		format2: {
+			message: "Convert the input format of 11/17/2020",
+			input: "11/17/2020",
+			expected: "2020-11-17"
+		},
+		format3: {
+			message: "Convert the input format of 17 November 2020",
+			input: "17 November 2020",
+			expected: "2020-11-17"
+		},
+		special1: {
+			message: "The input is undefined",
+			input: undefined,
+			expected: "NaN-NaN-NaN"
+		},
+		special2: {
+			message: "The input is empty",
+			input: "",
+			expected: "NaN-NaN-NaN"
+		},
+		special3: {
+			message: "The input format is invalid",
```

Comment:
Should add some more tests here to check marginal cases such as partially specified or otherwise out of range dates such as "20 November", "53 November", "November 2020" etc.